yaml.go

```diff
@@ -94,3 +106,35 @@ func buffer(file *os.File) []byte {
 	}
 	return make([]byte, size)
 }
+
+func isURL(pathname string) bool {
+    _, err := url.ParseRequestURI(pathname)
+    if err != nil {
+        return false
+    } else {
+        return true
+	}
+}
+
+func downloadFile(pathname string) (string, error) {
+	_, filename := path.Split(pathname)
+	tmpFile, err := ioutil.TempFile(os.TempDir(), filename)
```

Comment:
Instead of creating a temporary file here, you can make `downloadFile` write to the same `sink` channel as `readDir` and `readFile`.